raster-foundry/raster-foundry
# Add user scope migration and data model

## Overview

This PR adds a migration to add a not null jsonb `scopes` column to the `users` table with default values. It also creates data model for it.

### Checklist

- ~Styleguide updated, if necessary~
- ~Swagger specification updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

## Testing Instructions

- Run migrations
- Get a user ID
- In a Scala console, make sure below works:
```scala
import doobie._
import doobie.implicits._
import doobie.util.ExecutionContexts
import cats._
import cats.data._
import cats.effect.IO
import cats.implicits._
import io.circe.syntax._
import com.rasterfoundry.datamodel._
import com.rasterfoundry.database._
import com.rasterfoundry.database.util._
import com.rasterfoundry.database.Implicits._
import com.rasterfoundry.database.util.RFTransactor

val xa = RFTransactor.buildTransactor()
val userScopesIO = for {
  user <- UserDao.unsafeGetUserById("<user ID>")
  scopes <- sql"select scopes from users where id = '<user ID>'"
            .query[Map[ObjectType, List[ActionType]]]
            .unique
} yield { UserWithScopes(user, scopes) }
val userwithScopes = userScopesIO.transact(xa).unsafeRunSync
val jsonUserWithScope = userwithScopes.asJson
val jsonScope = userwithScopes.scopes.asJson
val userwithScopesFromJson = jsonUserWithScope.as[UserWithScopes]
val scopesFromJson = jsonScope.as[Map[ObjectType, List[ActionType]]]
```

Closes #4966
